### PR TITLE
Improve extraction of initials for avatars

### PR DESCRIPTION
**Pull request by rowanseymour**
Fixes https://github.com/nyaruka/rapidpro/issues/4341. A lot of workspaces are currently seeing weird initials, e.g. `UNICEF - Ireland` becomes `U-`

### CODE REVIEW

**Review by norkans7 — APPROVED** [review]

File: test/temba-menu.test.ts [hunk 1]
```diff
@@ -116,3 +117,17 @@ describe('temba-menu', () => {
     await assertScreenshot('menu/menu-refresh-2', getClip(menu));
   });
 });
+
+describe('avatars', () => {
+  it('can generate initials from text', async () => {
+    assert.equal(extractInitials(''), '?');
+    assert.equal(extractInitials('X'), 'X');
+    assert.equal(extractInitials('Acme'), 'AC');
+    assert.equal(extractInitials('al-Jazeera News'), 'AJ');
```
Comment (norkans7) [inline test/temba-menu.test.ts:126]: How will `U-Report Nigeria` and `U-Report Ghana` both going to be `UR`?
I think we want them as `UN` and `UG`

File: src/utils/index.ts [hunk 2]
```diff
@@ -666,6 +666,38 @@ export const getFullName = (user: User) => {
...
+ */
+export const extractInitials = (text: string) => {
+  // for empty or single chars, use placeholder or that char
+  if (text.length == 0) {
+    return '?';
+  } else if (text.length == 1) {
+    return text.toUpperCase();
+  }
+
+  const words = text.match(/\w+/g);
```
Comment (norkans7) [inline src/utils/index.ts:682]: we could use `[\w-]+` and ignore - later to never be returned as output

**Review by norkans7 — CHANGES_REQUESTED** [review]
hyphen should be considered when splitting words as a word character

File: test/temba-menu.test.ts (hunk 1 above)
Comment (rowanseymour) [reply test/temba-menu.test.ts]: Good point! Have tweaked and added support for other alphabets

**Review by norkans7 — APPROVED** [review]

**Pull request merged**